scripts/test/Muon/loading_tab/loadwidget_presenter_test.py

```diff
@@ -0,0 +1,358 @@
+    def mock_loading_from_browse(self, workspace, filename, run):
+        self.load_file_view.show_file_browser_and_return_selection = mock.Mock(
+            return_value=[filename])
+        self.load_file_model.load_workspace_from_filename = mock.Mock(
+            return_value=(workspace, run))
+
+    def mock_loading_from_current_run(self, workspace, filename, run):
+        self.load_run_model.load_workspace_from_filename = mock.Mock(
+            return_value=(workspace, filename, run))
+
+    def mock_user_input_single_run(self, workspace, filename, run):
+        self.load_run_view.get_run_edit_text = mock.Mock(return_value=str(run))
+        self.load_run_model.load_workspace_from_filename = mock.Mock(
+            return_value=(workspace, filename, run))
+
+    def mock_user_input_single_file(self, workspace, filename, run):
+        self.load_file_model.load_workspace_from_filename = mock.Mock(
+            return_value=(workspace, run))
+        self.load_file_view.get_file_edit_text = mock.Mock(
+            return_value=filename)
+
+    def mock_disabling_buttons_in_run_and_file_widget(self):
+        self.load_run_view.disable_load_buttons = mock.Mock()
+        self.load_run_view.enable_load_buttons = mock.Mock()
+        self.load_file_view.disable_load_buttons = mock.Mock()
+        self.load_file_view.enable_load_buttons = mock.Mock()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS : The interface should correctly load/display data via the run widget (load current run and user input run)
+    # and via the file widget (browse button or user enters file)
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_that_loading_single_file_via_browse_sets_model_data(self):
+        self.presenter.load_file_widget.on_browse_button_clicked()
+        self.Runner(self.presenter.load_file_widget._load_thread)
+
+        self.assertEqual(self.presenter._model.filenames, ["C:\dir1\dir2\dir3\EMU0001234.nxs"])
+        self.assertEqual(self.presenter._model.workspaces, [[1]])
+        self.assertEqual(self.presenter._model.runs, [1234])
+
+    def test_that_loading_single_file_via_browse_sets_run_and_file_view(self):
+        self.presenter.load_file_widget.on_browse_button_clicked()
+        self.Runner(self.presenter.load_file_widget._load_thread)
+
+        self.assertEqual(self.load_file_view.get_file_edit_text(), "C:\dir1\dir2\dir3\EMU0001234.nxs")
+        self.assertEqual(self.load_run_view.get_run_edit_text(), "1234")
+
+    def test_that_loading_via_browse_disables_all_load_buttons(self):
+        self.mock_disabling_buttons_in_run_and_file_widget()
+
+        self.presenter.load_file_widget.on_browse_button_clicked()
+        self.Runner(self.presenter.load_file_widget._load_thread)
+
+        self.assertEqual(self.load_file_view.disable_load_buttons.call_count,
```

Comment:
we should also check the value is 1 (if they are both zero the test should fail but will pass)

Reply:
have done